### ISSUE TITLE
Сайт: настроить автоматический бекап

### ISSUE THREAD

**Issue by Lady3mlnm** [TechnoFractal/calculator#9]
Возможны два пути:
1. Средствами провайдера VDS - Оля
2. Средствами самого Drupal-а - Мальвина, Аня
Совмещение данных способов приветствуется, посколько увеличит стабильность и послужит образовательным целям.

**Comment by olgapshen** [comment]
Тут есть еще уровень - бекап внутри операционной системы.
С VDS проблемно, так как это доп оплата.
Можно сохранять бекапы и выгружать их к нам даже на компьютеры.
Я налажу автобекап файлов сайта и БД.
Скорее всего это же можно сделать через Drupal.
Сейчас посмотрю.

**Comment by olgapshen** [comment]
Установила модуль backup_migrate.
Пойду дальше )

**Comment by olgapshen** [comment]
Не получается его найти, и я вижу ошибку:
![image](https://user-images.githubusercontent.com/13800244/28283945-2991efe6-6b38-11e7-836a-79d695408f52.png)

**Comment by olgapshen** [comment]
Он его установил но не дал права апачи на чтение..
![image](https://user-images.githubusercontent.com/13800244/28284020-5c7f18f2-6b38-11e7-8da8-439253556cab.png)
Это надо будет в FTP сконфигурировать

**Comment by olgapshen** [comment]
Должно исправить
![image](https://user-images.githubusercontent.com/13800244/28284152-b722333e-6b38-11e7-8f6c-f64ecc4c04b4.png)

**Comment by olgapshen** [comment]
Сработало!!!
![image](https://user-images.githubusercontent.com/13800244/28284271-0f8f0e52-6b39-11e7-9748-d89485f66a2c.png)

**Comment by olgapshen** [comment]
![image](https://user-images.githubusercontent.com/13800244/28284321-3c3dcbb4-6b39-11e7-9751-ef08e5d9c347.png)

После установки )

**Comment by olgapshen** [comment]
ОК, сайт можно бекапить.
Перед изминениями просто надо скачать бекап.
Завтра я локально подниму сайт и клонирую его через бекап.
http://technofractal.org/admin/config/development/backup_migrate
Далее это уже настройка бекапов по крону.
Это через другой плагин.
Настрою потом.

**Comment by olgapshen** [comment]
Пока задачу не закрываем.

**Comment by L3M-2** [comment]
Правильно, что не закрываем. Бекап через модуль Backup and Migrate по умолчанию не работает. Как поняла, нужно в настройках сайта прописать 'Private file system path'. В комментариях самого setting.php упомянуто, что путь должен быть абсолютным. Однако при комбинациях
```
$settings['file_private_path'] = 'http://www.technofractal.org/sites/default/files/private';
$settings['file_private_path'] = 'www.technofractal.org/sites/default/files/private';
$settings['file_private_path'] = 'technofractal.org/sites/default/files/private';
```
бекап делать отказывается.

Лишь при
`$settings['file_private_path'] = 'sites/default/files/private';`
можно создать бекап 'Default Drupal Database'. Но не 'Entire Site'.

Обращу внимание, что модуль Backup and Migrate находится в стадии разработки alpha2. В данный момент файл setting.php возвращён с своему первоначальному состоянию.
